i18n4k-generator/src/main/kotlin/de/comahe/i18n4k/generator/MessagesDataBundle.kt

```diff
@@ -41,8 +41,17 @@ class MessagesDataBundle(
                     .map { locale ->
                         messageFormatter.getMessageParametersNames(message, locale)
                     }
-            }.flatten()
-            .fold(mutableSetOf<CharSequence>()) { result, current ->
-                result.addAll(current); result
+            }
+            .flatten()
+            .flatten()
+            .fold(mutableMapOf<String, String?>()) { result, current ->
+                val split = current.split(Regex(":\\s*"))
+                val name = split[0]
+                val type = split.getOrNull(1)
+                if (name in result) {
+                    if (type != null && result[name] == null)
+                        result[name] = type
+                } else result[name] = type
+                result
```

Comment:
This is only needed on `addFactoryClass`. So this splitting of type inside the parameter name should be done there.

Reply:
Another idea: Parameter types may also be used in the normal LocalizedStringFactory via generics. So the messageFormatter should return a list of names and a types (as new class, not a map). Than this list of parameters can also be sorted, which is not so easy with a map.

I you don't how to do it, I will implement it, after this PR is merged.